semantic_router/llms/openai.py

```diff
@@ -36,22 +37,70 @@ def __init__(
         self.temperature = temperature
         self.max_tokens = max_tokens
 
-    def __call__(self, messages: List[Message]) -> str:
+    def _extract_tool_calls_info(self, tool_calls: list[dict[str, Any]]) -> list[dict[str, Any]]:
+        tool_calls_info = []
+        for tool_call in tool_calls:
+            if tool_call.function.arguments is None:
+                raise ValueError(
+                    "Invalid output, expected arguments to be specified for each tool call."
+                )
+            tool_calls_info.append({
+                "function_name": tool_call.function.name,
+                "arguments": tool_call.function.arguments
+            })
+        return json.dumps(tool_calls_info)
+    
+    def __call__(
+        self,
+        messages: List[Message],
+        function_schemas: Optional[list[dict[str, Any]]] = None,
+    ) -> str:
         if self.client is None:
             raise ValueError("OpenAI client is not initialized.")
         try:
+            if function_schemas:
+                tools = function_schemas
+            else:
+                tools = None
+
             completion = self.client.chat.completions.create(
                 model=self.name,
                 messages=[m.to_openai() for m in messages],
                 temperature=self.temperature,
                 max_tokens=self.max_tokens,
+                tools=tools,  # type: ignore # MyPy expecting Iterable[ChatCompletionToolParam] | NotGiven, but dict is accepted by OpenAI.
```

Comment:
We should pass the types that OpenAI expects here, rather than setting `tools = None` above we should be setting `tools = NotGiven` (we use the same elsewhere in the code)

Reply:
Cool no problem.

Do you want me to change `tools = list[dict[str,Any]` to `tools = Iterable[ChatCompletionToolParam]` too? or is that overkill?

Reply:
I think `list[dict[str, Any]` is fine — you will probably need to make that `List[Dict[str, Any]]` however as the lowercase `list[dict[...]]` were only supported from python 3.10 onwards (and we support 3.9 onwards here)

Reply:
Using `tools = NotGiven` now.